src/engine/cachingreader/cachingreader.cpp

```diff
@@ -79,6 +79,7 @@ void CachingReader::freeChunk(CachingReaderChunkForOwner* pChunk) {
     DEBUG_ASSERT(pChunk->getState() != CachingReaderChunkForOwner::READ_PENDING);
 
     const int removed = m_allocatedCachingReaderChunks.remove(pChunk->getIndex());
+    Q_UNUSED(removed);
```

Comment:
put this closer to the debug_assert, and add a comment that in the release build the assert is compiled away.  (and I think it makes sense to do this everywhere it happens, since it's only 5 instances